src/metadata.rs

```diff
@@ -183,7 +358,7 @@ impl Metadata21 {
         add_vec("Supported-Platform", &self.supported_platform);
         add_vec("Platform", &self.platform);
         add_vec("Supported-Platform", &self.supported_platform);
-        add_vec("Classifiers", &self.classifiers);
+        add_vec("Classifier", &self.classifiers);
```

Comment:
Unfortunately not: #407 #416. Should have definitely written a comment there though

Reply:
Actually it's only the PyPI upload api expects `classifiers`, not the wheel metadata itself I think.

twine uses the [pkginfo](https://github.com/pombredanne/pkginfo) package to [parse package metadata](https://github.com/pypa/twine/blob/7acd94e6e8e9a4ee9708526d826885ccf15afaa0/twine/wheel.py#L20-L37), pkginfo transforms `Classifier` into `classifiers` here: https://github.com/pombredanne/pkginfo/blob/4c69e7d22303f1f81f98165e98fc5ac13cea3c3e/pkginfo/distribution.py#L41

So what #416 did was right for the PyPI upload api but actually wrong for package metadata itself.

Reply:
I'd like to get this one right since other tools might expect `Classifier` for example PyOxidizer: https://github.com/indygreg/PyOxidizer/blob/1f0a9271568b187fd37867bcd76c4a378e49f5af/python-packaging/src/licensing.rs#L162

Reply:
Oh you're right, this is even more complex than thought